### DISCUSSION TITLE
Z Homing Failed

### DISCUSSION THREAD

**Discussion by vidiSetyanto** [Jyers/Marlin#1250] [Q&A] [answered]
Hi, I tried to download source code on the latest release. then i installed CRtouch.
I copied the setting from examples files for Ender 3 v2. 
then i change some parameter on configuration.h to enable bltouch.
#define BLTOUCH
  //#define MAX_SOFTWARE_ENDSTOP_Z
#define AUTO_BED_LEVELING_BILINEAR
#define Z_SAFE_HOMING
#define MULTIPLE_PROBING 2
#define MULTIPLE_PROBING 2
#define RESTORE_LEVELING_AFTER_G28

it can compiled  then uploaded to the printer.
somehow the printer cannot homing on z axis.
is there something that i missed?

* already unpluge z switch, when i downloaded and use precompiled firmware. it just worked fine.
* E3V2-BLTouch-5x5-HS-v4.2.2-v1.3.5.bin.
* i want to compile my self because want to change the max nozzle temp setting.
* 
here's the video:
https://user-images.githubusercontent.com/4694400/127740880-db3d67e3-47f6-4053-932c-cddba8da196c.mp4

**Comment by adam3654** [comment, marked as answer]
uncomment `#define USE_PROBE_FOR_Z_HOMING` in Configuration.h if you unplugged the z end stop limit switch. it's on or near line 1024

**Reply by vidiSetyanto** [reply]
seems doesn't work. still behave the same.

here my full config
[Marlin.zip](https://github.com/Jyers/Marlin/files/6911391/Marlin.zip)

**Reply by vidiSetyanto** [reply]
nvm, disabled //#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN. and working fine now. :)
